File: PRACTICA1/QytetetJava/src/modeloqytetet/Sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package modeloqytetet;

/**
 *
 * @author Jesus Manuel Garcia Palma
 * @correo xexugarcia95@gmail.com / jesusluisbcrespo@correo.ugr.es
 */
public class Sorpresa {
    private String texto;
    private TipoSorpresa tipo;
    int valor;

    public Sorpresa(String texto, TipoSorpresa tipo, int valor) {
        this.texto = texto;
        this.tipo = tipo;
        this.valor = valor;
    }

    private String getTexto() {
        return texto;
    }

    private TipoSorpresa getTipo() {
        return tipo;
    }

    private int getValor() {
        return valor;
    }

    @Override
    public String toString() {
        return "Sorpresa{" + "texto=" + texto + ", tipo=" + tipo + ", valor=" + Integer.toString(valor) + '}';
    }
    
    
}

File: PRACTICA1/QytetetRuby/lib/sorpresa.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "tipo_sorpresa.rb"
module ModeloQytetet
  class Sorpresa
    attr_reader :texto, :tipo, :valor
  
    def initialize(texto,tipo,valor)
      @texto = texto
      @tipo = tipo
      @valor = valor
    end
    
    def to_s
      "Texto: #{@texto} \n Valor: #{@valor} \n Tipo: #{@tipo}"
    end
  end
end